### PR TITLE
OOP School Library: Unit Tests

### PR DESCRIPTION
**Pull request by AliShahSafdari**
In this milestone we:

- [ ] Prepared tests for the Classes:
   `Book`
`Person`
`Student`
`Teacher`
`Classroom`
`Rental`
- [ ] Prepared tests for decorators

### CODE REVIEW

File: spec/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  context 'we create the book object and test it ' do
+    let(:book) { Book.new('C++', 'Safdari') }
+    it 'Book title must be C++' do
+      expect(book.title).to eq('C++')
+    end
+    it 'Book author must be C++' do
+      expect(book.author).to eq('Safdari')
+    end
+  end
+end
```
Comment (mckent05) [inline spec/book_spec.rb]: - Good job done so far, but according to the project requirements, you are required to have unit tests for all the methods in all the classes. Please add tests for the `add_rental` method in your `Book` class 😄

File: spec/person_spec.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    end
+    it 'Perosn parent_permission must be true' do
+      expect(person.parent_permission).to be true
+    end
+
+    it 'Teacher can_use_services must be true' do
+      expect(person.can_use_services?).to eq(true)
+    end
+  end
+end
```
Comment (mckent05) [inline spec/person_spec.rb]: - Good job done so far, but according to the project requirements, you are required to have unit tests for all the methods in all the classes. Please add tests for the `add_rental` method in your `Person` class 😄

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @AliShahSafdari, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job adding unit tests to your application 👍 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/person_spec.rb (hunk 2 above)
Comment (AliShahSafdari) [reply spec/person_spec.rb]: Thanks, dear Engineer, we will add as you mentioned.

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @AliShahSafdari, and @Nasirkhan294, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

👍 You did well to add tests for `add_rental` methods as requested by the previous reviewer.
👏 App the tests pass.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**